src/source/source_cache.js

```diff
@@ -460,20 +461,21 @@ class SourceCache extends Evented {
      * Removes tiles that are outside the viewport and adds new tiles that
      * are inside the viewport.
      * @private
-     * @param {boolean} updateForTerrain Signals to update tiles even if the
-     * source is not used (this.used) by layers: it is used for terrain.
-     * @param {tileSize} tileSize If needed to get lower resolution ideal cover,
-     * override source.tileSize used in tile cover calculation.
      */
-    update(transform: Transform, tileSize?: number, updateForTerrain?: boolean) {
+    update(transform: Transform) {
```

Comment:
Is this change orthogonal to the 404 feature implementation? If so, I'd suggest to keep it out.

The issue with the change is that it might be breaking current approach, and it is good to check:

Terrain source cache is updated from two places:

a) updating all source caches (when called from there, L472 `if (this.usedForTerrain && !updateForTerrain) { ` prevents update twice as updateForTerrain is false.
https://github.com/mapbox/mapbox-gl-js/blob/05ec03e6bacb332c82e6465cd7d43b7d74da1e77/src/style/style.js#L1637-L1641

b) update from terrain. There updateForTerrain is set to true and update passes L472 check. 

https://github.com/mapbox/mapbox-gl-js/blob/cd46ea1da423e74e56510bcd4c268041f000846f/src/terrain/terrain.js#L281

There, tile size is set and also updateForTerrain flag to true.

Reply:
Yes, this change is orthogonal to the 404 feature implementation, and it breaks the current approach. I've reverted this